expr/functions/moving/function_test.go

```diff
@@ -133,6 +133,14 @@ func TestMoving(t *testing.T) {
 			[]*types.MetricData{types.MakeMetricData("movingMax(metric1,2)",
 				[]float64{math.NaN(), math.NaN(), 2, 3, 3, 2}, 1, 0).SetTag("movingMax", "2").SetNameTag("movingMax(metric1,2)")}, // StartTime = from
 		},
+		{
+			"movingAverage(metric1,'3sec')", // Check that a window that consists only of NaN points returns a value of math.NaN()
+			map[parser.MetricRequest][]*types.MetricData{
+				{"metric1", -3, 1}: {types.MakeMetricData("metric1", []float64{1e20, 0.e-20, 1, math.NaN(), math.NaN(), math.NaN(), 1, 1, math.NaN(), math.NaN(), math.NaN(), math.NaN(), 2, 1, 0}, 1, now32)},
```

Comment:
I think it's worth adding this note so the values of the test are not changed in the future
```suggestion
				// These values introduce numerical errors in the sum, making it non-zero when it should be. 
				// This causes a `0.0000...01/0` division, that results in infinity
				{"metric1", -3, 1}: {types.MakeMetricData("metric1", []float64{1e20, 0.e-20, 1, math.NaN(), math.NaN(), math.NaN(), 1, 1, math.NaN(), math.NaN(), math.NaN(), math.NaN(), 2, 1, 0}, 1, now32)},
```